neurodsp/sim/periodic.py

```diff
@@ -104,62 +126,186 @@ def sim_bursty_oscillation(n_seconds, fs, freq, enter_burst=.2, leave_burst=.2,
+        Definition of whether each cycle is bursting or not.
+    cycle : 1d array
+        The cycle to use for bursts.
+
+    Returns
+    -------
+    burst_sig : 1d array
+        Simulated bursty oscillation.
+    """
+
+    n_samples = int(n_seconds * fs)
+    n_samples_cycle = len(cycle)
+
+    burst_sig = np.zeros([n_samples])
+    for sig_ind, is_osc in zip(range(0, n_samples, n_samples_cycle), is_oscillating):
+        if is_osc:
+            burst_sig[sig_ind:sig_ind+n_samples_cycle] = cycle
 
-    for ii in range(1, n_cycles):
+    return burst_sig
+
+
+def make_is_osc_prob(n_cycles, enter_burst, leave_burst):
+    """Create bursting definition, based on probabilistic burst starts and stops.
+
+    Parameters
+    ----------
+    n_cycles : int
+        The number of cycles to simulate the burst definition for.
+    enter_burst : float, optional, default: 0.2
+        Probability of a cycle entering a burst, given the last cycle is not oscillating.
+    leave_burst : float, optional, default: 0.2
+        Probability of a cycle leaving a burst, given the last cycle is oscillating.
+
+    Returns
+    -------
+    is_oscillations : 1d array of bool
+        Definition of whether each cycle is bursting or not.
+    """
+
+    check_param(enter_burst, 'enter_burst', [0., 1.])
+    check_param(leave_burst, 'leave_burst', [0., 1.])
+
+    # Initialize vector of burst definitions
+    is_oscillating = np.array([False] * (n_cycles))
+
+    for ind in range(1, n_cycles):
 
         rand_num = np.random.rand()
 
-        if is_oscillating[ii-1]:
-            is_oscillating[ii] = rand_num > leave_burst
+        # If prior cycle bursting, leave burst with given probability
+        if is_oscillating[ind-1]:
+            is_oscillating[ind] = rand_num > leave_burst
+        # Otherwise, with prior cycle not bursting, enter burst with given probability
         else:
-            is_oscillating[ii] = rand_num < enter_burst
+            is_oscillating[ind] = rand_num < enter_burst
+
+    return is_oscillating
+
+
+def make_is_osc_durations(n_cycles, n_cycles_burst, n_cycles_off):
+    """Create bursting definition, based on cycle lengths and intervals.
+
+    Parameters
+    ----------
+    n_cycles : int
+        The number of cycles to simulate the burst definition for.
+    n_cycles_burst : int
+        Number of cycles within a burst burst.
+    n_cycles_off : int
+        Number of cycles between bursts.
+
+    Returns
+    -------
+    is_oscillations : 1d array of bool
+        Definition of whether each cycle is bursting or not.
+    """
+
+    # Make the burst parameters iterators
+    n_cycles_burst = repeat(n_cycles_burst) if isinstance(n_cycles_burst, int) else n_cycles_burst
+    n_cycles_off = repeat(n_cycles_off) if isinstance(n_cycles_off, int) else n_cycles_off
+
+    # Initialize is oscillating
+    is_oscillating = np.array([False] * (n_cycles))
+
+    # Set the first bursting index
+    ind = 1
```

Comment:
yeh... this is sorta a legacy thing. By convention, the old (prob) approach was defined to start off-burst (have to start somewhere). I kept this convention for now, but it makes perhaps less sense in the `durations` approach. Perhaps it makes sense in `durations` to define if it starts in or out of a burst, rather than the more arbitrary 'start with one non-burst cycle, then into a burst'

Reply:
okay, for `durations` I've set the default to start at zero, on a burst. I can't think of anything better to do. I'm not sure if there is an easy way to toggle whether there is an easy way to toggle whether it starts on or off burst.

Reply:
Sounds good!

I'm not sure either. If np.tile could be used, the order of what's being tiled could be flipped to toggle starting with an on/off burst. But variable durations from a generator input makes this a bit more difficult.

Reply:
yeh, so I think I'm just going to keep it like this for now, since there isn't anything else that seems clear to do.